### PR TITLE
fix: [PLG-1086]: Credit Card Flow

### PR DESCRIPTION
**Pull request by devinder-harness**
### Summary

This PR adds new UI updates , bug fixes and polling to current credit card flow

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: configs/restful-react.config.js [hunk 1]
```diff
@@ -32,9 +32,10 @@ module.exports = {
       base: `{getConfig("ng/api")}`
     },
     customGenerator: arg => customGenerator(arg, "getConfig('ng/api')"),
-    ...(process.env.cdng_schema_path
```
Comment (vkbansal-harness) [inline configs/restful-react.config.js:35]: revert this

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 297 Issues 
* Code issue 297 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10729)

File: src/framework/LicenseStore/LicenseStoreContext.tsx [hunk 2]
```diff
@@ -200,18 +201,18 @@ export function LicenseStoreProvider(props: React.PropsWithChildren<unknown>): R
...
         }, POLL_VERSION_INTERVAL)
       } catch (_err) {
-        clearTimeout(getVersionTimeOut)
+        clearTimeout(timerRef.current)
       }
     }
 
     return () => {
-      clearTimeout(getVersionTimeOut)
+      clearTimeout(timerRef.current)
```
Comment (vkbansal-harness) [inline src/framework/LicenseStore/LicenseStoreContext.tsx:215]:
```suggestion
      if (timerRef.current) {
        clearTimeout(timerRef.current)
      }
```

File: src/framework/LicenseStore/LicenseStoreContext.tsx [hunk 3]
```diff
@@ -200,18 +201,18 @@ export function LicenseStoreProvider(props: React.PropsWithChildren<unknown>): R
             versionMap: response.data || {}
           }))
         }
-
+        clearTimeout(timerRef.current)
```
Comment (vkbansal-harness) [inline src/framework/LicenseStore/LicenseStoreContext.tsx:204]:
```suggestion
    if (timerRef.current)
        clearTimeout(timerRef.current)
      }
```

File: src/framework/LicenseStore/LicenseStoreContext.tsx [hunk 4]
```diff
@@ -200,18 +201,18 @@ export function LicenseStoreProvider(props: React.PropsWithChildren<unknown>): R
...
         // set next poll
-        getVersionTimeOut = setTimeout(() => {
+        timerRef.current = window.setTimeout(() => {
           pollVersionMap(latestVersionMap || versionMap)
         }, POLL_VERSION_INTERVAL)
       } catch (_err) {
-        clearTimeout(getVersionTimeOut)
+        clearTimeout(timerRef.current)
```
Comment (vkbansal-harness) [inline src/framework/LicenseStore/LicenseStoreContext.tsx:210]:
```suggestion
    if (timerRef.current) {
        clearTimeout(timerRef.current)
      }
```

File: src/modules/10-common/hooks/useRegionList.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      setStates({ loading: false, list: stateList })
+    })
+  }, [])
+  return countries.loading || states.loading
+    ? { countries: [], states: {}, loading: true }
+    : { countries: countries.list, states: states.list, loading: false }
+}
+
+export const fetchCountries = async (): Promise<Country[]> => {
+  const countries = await import('./countries.json')
```
Comment (vkbansal-harness) [inline src/modules/10-common/hooks/useRegionList.tsx:50]: 🔥

File: src/modules/10-common/hooks/useRegionList.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  return countries.default
+}
+
+export const fetchStates = async (): Promise<StateByCountryMap> => {
+  const states = (await import('./statesByCountry.json')) as unknown as { default: StateByCountryMap }
```
Comment (vkbansal-harness) [inline src/modules/10-common/hooks/useRegionList.tsx:55]: 🔥

File: src/modules/10-common/layouts/TrialLicenseBanner.tsx [hunk 7]
```diff
@@ -142,8 +142,9 @@ const ExtendOrFeedBackBtn: React.FC<{
   const { accountId } = useParams<AccountPathProps>()
   const { trackEvent } = useTelemetry()
   const { showError } = useToaster()
-  const { licenseInformation, updateLicenseStore } = useLicenseStore()
-
+  // const { licenseInformation, updateLicenseStore } = useLicenseStore()
```
Comment (vkbansal-harness) [inline src/modules/10-common/layouts/TrialLicenseBanner.tsx:145]: intentional?

File: src/modules/33-auth-settings/components/Subscription/CostCalculator/FFMAUCard.tsx [hunk 8]
```diff
@@ -6,39 +6,63 @@
...
+  selectedVal: number,
+  _list: number[],
+  minValue: number,
+  _stepSize: number
+): number => {
+  const finalValue = selectedVal
+  if (selectedVal === 0) {
+    return minValue
+  }
+  // if (!list.includes(finalValue)){
```
Comment (vkbansal-harness) [inline src/modules/33-auth-settings/components/Subscription/CostCalculator/FFMAUCard.tsx:31]: remove commented code

File: src/modules/33-auth-settings/components/Subscription/FinalReview/SubscriptionDetailsCard.tsx [hunk 9]
```diff
@@ -8,17 +8,31 @@
...
 }
-const SubscriptionDetailsCard: React.FC<SubscriptionDetailsCardProps> = ({ newPlan, items, setView }) => {
+const SubscriptionDetailsCard: React.FC<SubscriptionDetailsCardProps> = ({
+  newPlan,
+  items,
+  setView
+  // subscriptionId
+}) => {
   const { getString } = useStrings()
+  // const { accountId } = useParams<AccountPathProps>()
```
Comment (vkbansal-harness) [inline src/modules/33-auth-settings/components/Subscription/FinalReview/SubscriptionDetailsCard.tsx:31]: remove commented code

File: src/modules/33-auth-settings/components/Subscription/PaymentMethod/SubscriptionPoll.tsx.new [hunk 10]
```diff
@@ -0,0 +1 @@
+/*
```
Comment (vkbansal-harness) [inline src/modules/33-auth-settings/components/Subscription/PaymentMethod/SubscriptionPoll.tsx.new:1]: intended?

**Comment by devinder-harness** [comment]
retrigger titlecheck

**Comment by devinder-harness** [comment]
dockerize

**Review by vardanbansal-harness — APPROVED** [review]

**Review by reetika04 — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Comment by kapilkumawat86** [comment]
FM on request

**Pull request merged**